src/amqp.js

```diff
@@ -0,0 +1,340 @@
+    debug(`Purging queue: ${queue}`);
+    try {
+      errorUndefinedQueue(queue);
+      const channelInfo = await channel.assertQueue(queue, {durable: true});
+      await channel.purgeQueue(queue);
+      debug(`Queue ${queue} has purged ${channelInfo.messageCount} messages`);
+
+      return true;
+    } catch (error) {
+      handleAmqpErrors(error);
+    }
+  }
+
+  // MARK: Count queue
+  /**
+   * Count queue uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @returns {number} returns number of messages in queue
+   */
+  async function countQueue({blobId, status}) {
+    const queue = generateQueueId({blobId, status});
+    debug(`Counting queue: ${queue}`);
+    try {
+      errorUndefinedQueue(queue);
+      const channelInfo = await channel.assertQueue(queue, {durable: true});
+      return channelInfo.messageCount;
+    } catch (error) {
+      handleAmqpErrors(error);
+    }
+  }
+
+  // MARK: Get chunk
+  /**
+   * Gets 100 messages from queue. CHUNK_SIZE set in constants.
+   * Get chunk uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @returns {object|boolean} returns {headers, records, messages} or false
+   */
+  async function getChunk({blobId, status}) {
+    const queue = generateQueueId({blobId, status});
+    try {
+      errorUndefinedQueue(queue);
+      const channelInfo = await channel.assertQueue(queue, {durable: true});
+      if (checkMessageCount(channelInfo)) {
+        debug(`Prepared to getChunk from queue: ${queue}`);
+        // getData: next chunk (100) messages
+        const messages = await getData(queue);
+
+        // Note: headers are from the first message of the chunk
+        debug(`getChunk (${messages ? messages.length : '0'} from queue ${queue}) to records`);
+
+        const headers = getHeaderInfo(messages[0]);
+        const records = await messagesToRecords(messages);
+        return {headers, records, messages};
+      }
+
+      return false;
+    } catch (error) {
+      handleAmqpErrors(error);
+    }
+  }
+
+  // MARK: Get one
+  /**
+   * Gets 1 messages from queue.
+   * Get one uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @returns {object|boolean} returns {headers, records, messages} or false
+   */
+  async function getOne({blobId, status}) {
```

Comment:
Function name could reference to a queue message so it would be clear based on name, that this gets one queue message.

Reply:
If we update getOne to getOneMessage or something, we'll probably want to update getChunk to getChunkOfMessages or something